src/CiviCrmEntityViewBuilder.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Drupal\civicrm_entity;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityViewBuilder;
+
+class CiviCrmEntityViewBuilder extends EntityViewBuilder {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getBuildDefaults(EntityInterface $entity, $view_mode) {
+    $defaults = parent::getBuildDefaults($entity, $view_mode);
+
+    // Set a default theme.
+    $defaults['#theme'] = 'civicrm_entity';
```

Comment:
This is required because we're not generating theme hooks (and ergo template files) for each discoverable entity type. The parent logic contains

```
    // Add the default #theme key if a template exists for it.
    if ($this->themeRegistry->getRuntime()->has($this->entityTypeId)) {
      $build['#theme'] = $this->entityTypeId;
    }
```

So +1 to this, it standardizes the theme hook for all CiviCRM entities.

Reply:
I like this too, wish I had done it this way for D7....one standard file, that is over-ridable per entity type by template name..less files to maintain. Might just "backport" this.